File: Data Structures & Algorithms/missing-number/submission-3.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length; // The size of the array
        int ans = 0;

        // XOR all numbers from 0 to n
        for (int i = 0; i <= n; i++) {
            ans ^= i; // XOR all numbers from 0 to n
        }

        // XOR all numbers in the array
        for (int num : nums) {
            ans ^= num; // XOR all numbers in nums
        }

        // The result is the missing number
        return ans;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        L = 0
        R = len(numbers) - 1
        total = numbers[L] + numbers[R]

        while L < R and total != target:
            
            while target > total and L < R:
                L += 1
                total = numbers[L] + numbers[R]

            while target < total and L < R:
                R -= 1
                total = numbers[L] + numbers[R]

        return [L+1, R+1]


                

        
        